error[E0308]: mismatched types
    --> .../udevrs-0.3.0/src/monitor.rs:892:58
     |
892  |                 if unsafe { libc::poll(pfd.as_mut_ptr(), pfd_len, 0) } > 0 {
     |                             ----------                   ^^^^^^^ expected `u32`, found `u64`
     |                             |
     |                             arguments to this function are incorrect
     |

---

--- a/src/monitor.rs
+++ b/src/monitor.rs
@@ -889,7 +889,7 @@ impl UdevMonitor {
 
                 // SAFETY: call to `poll` is safe because `pollfd` is properly initialized, and the
                 // resulting mutable pointer references valid memory.
-                if unsafe { libc::poll(pfd.as_mut_ptr(), pfd_len, 0) } > 0 {
+                if unsafe { libc::poll(pfd.as_mut_ptr(), pfd_len.try_into().unwrap(), 0) } > 0 {
                     // retry with the next device
                     Ok(())
                 } else {
